### DISCUSSION TITLE
schema does not sync pluguin DB2

### DISCUSSION THREAD

**Discussion by antoniocarlosbotelho** [alisonrafael/metabase-db2-driver#7] [General]
Is it possible to synchronize the DB2 database so that we can see the tables, without having to do native queries?
today we still don't import the database into the metabase through the db2 plugin.

![image](https://user-images.githubusercontent.com/74840997/233475244-59f18598-c1ae-4a8c-9080-b57617f1f386.png)

**Comment by alisonrafael** [comment]
Hi, 
I hadn't noticed this problem. Table synchronization stopped working in v0.46. I will investigate what happened.

**Reply by antoniocarlosbotelho** [reply]
excellent thank you

**Reply by alisonrafael** [reply]
Hi, 
you are from Brazil?

I have updated driver version and its available [here](https://github.com/alisonrafael/metabase-db2-driver/releases/download/v1.1.46.1/db2.metabase-driver.jar)!

**Reply by antoniocarlosbotelho** [reply]
Show, friend it worked perfectly.

Yes, I speak of Brazil from Belém do Pará

![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/073fd901-6ca0-49e9-a6d3-63a39c4c1e98)

**Comment by antoniocarlosbotelho** [comment]
Good afternoon, in the variable field filter when filtering a table, with the ID values, the search does not pull the values to select the IDs

![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/2024db2d-d4c8-43b5-8b10-34fb1939a9e5)

here is a query on the specific example table
![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/8abb1ff9-e5a5-49bf-8adc-10fb6354ff70)

**Reply by alisonrafael** [reply]
Hi!

The error SQLCODE=-206 occurs because field filter doesn't work when you use alias in tables.
See here an explanation https://www.metabase.com/learn/sql-questions/field-filters#field-filters-are-incompatible-with-aliasing

Another thing is "_the search does not pull the values to select the IDs_". Maybe metabase do not did fingerprint of the table yet. You can try to force metabase to fingerprint an especific table doing this (https://www.metabase.com/docs/latest/troubleshooting-guide/sync-fingerprint-scan#fingerprinting):

1. Go to Admin > Data Model.
2. Select your database and table.
3. Change the visibility of the table to “Hidden”.
4. Change the visibility back to “Queryable”.
5. Wait 10 seconds.

Maybe an error occurs at that moment that you will be able to see in the log. 

Se você achar melhor, você pode me enviar um e-mail direto para mim alisonrafael@hotmail.com. Você está me ajudando bastante com os testes!

**Reply by antoniocarlosbotelho** [reply]
Really in the simple query without alias and I managed to filter, but I had to change the column to category just so it made my filter
![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/6c2d2640-35a6-4640-a80c-97e0f506b90d)

aqui foi feito as alterações 
![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/30cd41f3-8ca5-4758-8d88-9261f06909ed)


![image](https://github.com/alisonrafael/metabase-db2-driver/assets/74840997/8711a9f2-ff50-4917-aed9-4505c66f9b7c)
